### ISSUE TITLE
`jsx-a11y/role-supports-aria-props `: `role="none"` should support `aria-hidden`

### ISSUE THREAD

**Issue by Janpot** [jsx-eslint/eslint-plugin-jsx-a11y#1090]
`jsx-a11y/role-supports-aria-props` when run on a element with role `none` should behave exactly like role `presentation`. The former currently fails on `aria-hidden` attribute, while the latter passes.

Reproduction: https://stackblitz.com/edit/stackblitz-starters-hqkhlxwm?file=index.js

`aria-hidden` is a global ARIA state. Per the WAI-ARIA spec ([§6.5 Global States and Properties](https://www.w3.org/TR/wai-aria/#global_states))

> The following global states and properties are supported by all roles and by all base markup elements unless otherwise prohibited.
> * ...
> * [aria-hidden (state)](https://www.w3.org/TR/wai-aria/#aria-hidden)
> * ...

The only props prohibited on `presentation`/`none` are `aria-label` and `aria-labelledby` (added in ARIA 1.2 as "prohibited states and properties"). `aria-hidden` is not on that prohibited list, so it remains globally supported on these roles. [GitHub](https://github.com/w3c/aria/issues/1824)
And the spec defines `none` and `presentation` as synonyms
